# Covid Cased worldwide

Importing data to notebook

In [3]:
import pandas as pd

In [4]:
covid_df = pd.read_csv('covid19.csv')

Checking for the number of columns and rows

In [6]:
covid_df.shape

(10903, 14)

This means there is a total of 10,903 rowns and 14 columns

# Data Cleaning Process

For cleaning purposes, I am checking to see if any of the rows are missing data.

In [8]:
covid_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10903 entries, 0 to 10902
Data columns (total 14 columns):
 #   Column                   Non-Null Count  Dtype 
---  ------                   --------------  ----- 
 0   Date                     10903 non-null  object
 1   Continent_Name           10903 non-null  object
 2   Two_Letter_Country_Code  10903 non-null  object
 3   Country_Region           10903 non-null  object
 4   Province_State           10903 non-null  object
 5   positive                 10903 non-null  int64 
 6   hospitalized             10903 non-null  int64 
 7   recovered                10903 non-null  int64 
 8   death                    10903 non-null  int64 
 9   total_tested             10903 non-null  int64 
 10  active                   10903 non-null  int64 
 11  hospitalizedCurr         10903 non-null  int64 
 12  daily_tested             10903 non-null  int64 
 13  daily_positive           10903 non-null  int64 
dtypes: int64(9), object(5)
memory usage: 1

List of all columns to see if anything needs to be changed

In [24]:
covid_df.columns

Index(['Date', 'Continent_Name', 'Two_Letter_Country_Code', 'Country_Region',
       'Province_State', 'positive', 'hospitalized', 'recovered', 'death',
       'total_tested', 'active', 'hospitalizedCurr', 'daily_tested',
       'daily_positive'],
      dtype='object')

Created a dictionary to change 3 column names

In [25]:
covid_df.rename(columns={
    'Continent_Name': 'Continent',
    'Two_Letter_Country_Code': 'Country_Code',
    'Province_State': 'State'
}, inplace=True)

In [26]:
covid_df.columns

Index(['Date', 'Continent', 'Country_Code', 'Country_Region', 'State',
       'positive', 'hospitalized', 'recovered', 'death', 'total_tested',
       'active', 'hospitalizedCurr', 'daily_tested', 'daily_positive'],
      dtype='object')

This list was created to capitlize the first letter of each word

In [28]:
covid_df.columns = ['Date', 'Continent', 'Country_Code', 'Country_Region', 'State',
       'Positive', 'Hospitalized', 'Recovered', 'Death', 'Total_Tested',
       'Active', 'HospitalizedCurr', 'Daily_Tested', 'Daily_Positive']

In [30]:
covid_df.columns

Index(['Date', 'Continent', 'Country_Code', 'Country_Region', 'State',
       'Positive', 'Hospitalized', 'Recovered', 'Death', 'Total_Tested',
       'Active', 'HospitalizedCurr', 'Daily_Tested', 'Daily_Positive'],
      dtype='object')

In [36]:
covid_df.describe()

Positive  Hospitalized      Recovered         Death  Total_Tested  \
count  1.090300e+04  10903.000000   10903.000000  10903.000000  1.090300e+04   
mean   1.776802e+04    766.402550    2409.220490    947.592956  1.954757e+05   
std    9.314346e+04   5626.325766   11254.209177   5507.242206  8.190222e+05   
min    0.000000e+00      0.000000       0.000000      0.000000  0.000000e+00   
25%    4.400000e+01      0.000000       0.000000      0.000000  2.147500e+03   
50%    1.026000e+03      0.000000       0.000000      6.000000  3.035800e+04   
75%    7.440000e+03      0.000000     500.500000    136.000000  1.252850e+05   
max    1.783570e+06  89590.000000  171883.000000  98536.000000  1.693689e+07   

              Active  HospitalizedCurr   Daily_Tested  Daily_Positive  
count   10903.000000      10903.000000   10903.000000    10903.000000  
mean     4472.194717        454.074475    6841.403283      497.884619  
std     22277.479931       2253.045376   27198.457892     2354.834051  
min        -2.000000          0.000000 -330230.000000    -4735.000000  
25%         0.000000          0.000000      71.000000        1.000000  
50%         0.000000          0.000000    1048.000000       29.000000  
75%       660.500000         28.000000    4688.500000      264.000000  
max    280931.000000      33004.000000  492276.000000    63047.000000

In [38]:
covid_df['Continent'].describe()

count             10903
unique                6
top       North America
freq               6452
Name: Continent, dtype: object

Contient Column has 6 unique values, top value is North America, which shows up 6452 times.

In [43]:
covid_df.corr()

Positive  Hospitalized  Recovered     Death  Total_Tested  \
Positive          1.000000      0.224506   0.274848  0.927603      0.880175   
Hospitalized      0.224506      1.000000   0.310516  0.241926      0.082223   
Recovered         0.274848      0.310516   1.000000  0.330940      0.401057   
Death             0.927603      0.241926   0.330940  1.000000      0.789355   
Total_Tested      0.880175      0.082223   0.401057  0.789355      1.000000   
Active            0.314676      0.731876   0.646578  0.315093      0.344443   
HospitalizedCurr  0.190164      0.308061   0.378250  0.316667      0.119578   
Daily_Tested      0.841353      0.069533   0.314510  0.736571      0.932272   
Daily_Positive    0.790040      0.113165   0.132055  0.641467      0.667083   

                    Active  HospitalizedCurr  Daily_Tested  Daily_Positive  
Positive          0.314676          0.190164      0.841353        0.790040  
Hospitalized      0.731876          0.308061      0.069533        0.113165  
Recovered         0.646578          0.378250      0.314510        0.132055  
Death             0.315093          0.316667      0.736571        0.641467  
Total_Tested      0.344443          0.119578      0.932272        0.667083  
Active            1.000000          0.505717      0.333117        0.230089  
HospitalizedCurr  0.505717          1.000000      0.126723        0.182067  
Daily_Tested      0.333117          0.126723      1.000000        0.766903  
Daily_Positive    0.230089          0.182067      0.766903        1.000000